Logging/ARKLogDistributor.m

```diff
@@ -91,28 +97,43 @@ - (instancetype)init;
 
 - (ARKLogStore *)defaultLogStore;
 {
-    dispatch_once(&_defaultLogStoreAccessOnceToken, ^{
-        // Lazily create a default log store if none exists.
-        if (self.weakDefaultLogStore == nil) {
-            self.defaultLogStore = [[ARKLogStore alloc] initWithPersistedLogFileName:[NSStringFromClass([self class]) stringByAppendingString:@"_DefaultLogStore"]];
+    /**
+    * Ensure that changes to self.defaultLogStoreAccessorCalled and self.defaultLogStore (via self.weakDefaultLogStore)
+    * are mututally exclusive.
+    */
+    [self.defaultLogStorePropertyLock lock];
+    {
+        if (!self.defaultLogStoreAccessorCalled && self.weakDefaultLogStore == nil) {
+            // Lazily create a default log store if none exists.
+            ARKLogStore *defaultLogStore = [[ARKLogStore alloc] initWithPersistedLogFileName:[NSStringFromClass([self class]) stringByAppendingString:@"_DefaultLogStore"]];
+            defaultLogStore.name = @"Default";
+            defaultLogStore.prefixNameWhenPrintingToConsole = NO;
+            self.defaultLogStore = defaultLogStore;
         }
-    });
+
+        self.defaultLogStoreAccessorCalled = YES;
+    }
+    [self.defaultLogStorePropertyLock unlock];
     
     return self.weakDefaultLogStore;
 }
 
 - (void)setDefaultLogStore:(ARKLogStore *)logStore;
 {
-    // Remove the old log store.
-    [self removeLogObserver:self.weakDefaultLogStore];
-    
-    if (logStore) {
-        // Add the new log store. The logObserver array will hold onto the log store strongly.
-        [self addLogObserver:logStore];
+    [self.defaultLogStorePropertyLock lock];
```

Comment:
Technically this lock only needs to be around setting the `weakDefaultLogStore`, but I kinda dig having it around the whole process